Sources/CvPlot.cpp

```diff
@@ -970,81 +936,62 @@ void CvPlot::doImprovement()
 
 void CvPlot::doImprovementUpgrade()
 {
-	if (getImprovementType() != NO_IMPROVEMENT  && !isPlotIgnoringImprovementUpgrade())
+	const ImprovementTypes eImprovement = getImprovementType();
+	if (eImprovement != NO_IMPROVEMENT && !isPlotIgnoringImprovementUpgrade())
 	{
-/************************************************************************************************/
-/* Afforess	                  Start		 05/23/10                                               */
-/*                                                                                              */
-/*                                                                                              */
-/************************************************************************************************/
-/*
-		ImprovementTypes eImprovementUpdrade = (ImprovementTypes)GC.getImprovementInfo(getImprovementType()).getImprovementUpgrade();*/
-		ImprovementTypes eImprovement = getImprovementType();
-		ImprovementTypes eImprovementUpdrade = GET_TEAM(getTeam()).getImprovementUpgrade(getImprovementType());
-/************************************************************************************************/
-/* Afforess	                     END                                                            */
-/************************************************************************************************/
-		if (eImprovementUpdrade != NO_IMPROVEMENT && (GC.getImprovementInfo(eImprovementUpdrade).getPrereqTech() == NO_TECH || (GC.getImprovementInfo(eImprovementUpdrade).getPrereqTech() != NO_TECH && GET_TEAM(getTeam()).isHasTech((TechTypes)GC.getImprovementInfo(eImprovementUpdrade).getPrereqTech()))))
+		const CvTeam& team = GET_TEAM(getTeam());
+		const ImprovementTypes eImprovementUpgrade = team.getImprovementUpgrade(eImprovement);
+		if (eImprovementUpgrade != NO_IMPROVEMENT)
 		{
-			if (GC.getImprovementInfo(eImprovementUpdrade).getHighestCost() < GET_PLAYER(getOwnerINLINE()).getEffectiveGold())
+			const CvImprovementInfo& upgradeInfo = GC.getImprovementInfo(eImprovementUpgrade);
+			if ((upgradeInfo.getPrereqTech() == NO_TECH || team.isHasTech((TechTypes)upgradeInfo.getPrereqTech()))
+				&& upgradeInfo.getHighestCost() < GET_PLAYER(getOwner()).getEffectiveGold())
 			{
-				// Super Forts begin *upgrade* - added if-else statement
-				if(GC.getImprovementInfo(getImprovementType()).isUpgradeRequiresFortify())
+				// Super Forts *upgrade*
+				if (GC.getImprovementInfo(eImprovement).isUpgradeRequiresFortify())
 				{
-					bool bDefenderFound = false;
 					CLinkList<IDInfo> oldUnits;
 					CLLNode<IDInfo>* pUnitNode = headUnitNode();
-					CvUnit* pLoopUnit;
-					
 
 					while (pUnitNode != NULL)
 					{
 						oldUnits.insertAtEnd(pUnitNode->m_data);
 						pUnitNode = nextUnitNode(pUnitNode);
 					}
 
+					bool bDefenderFound = false;
 					pUnitNode = oldUnits.head();
```

Comment:
This is the same as `headUnitNode()` isn't it?! Why is it using the `oldUnits.head()` but then using the normal `nextUnitNode` function to get the remaining ones? 
WTF was the author thinking ?!